modules/virtual_network_gateway/variables.tf

```diff
@@ -91,39 +89,62 @@ variable "edge_zone" {
 }
 
 variable "enable_bgp" {
-  description = "If true, BGP (Border Gateway Protocol) will be enabled for this Virtual Network Gateway. Defaults to false"
+  description = "Controls whether BGP (Border Gateway Protocol) will be enabled for this Virtual Network Gateway."
   default     = false
+  nullable    = false
   type        = bool
 }
 
 variable "generation" {
-  description = "The Generation of the Virtual Network gateway. Possible values include Generation1, Generation2 or None"
+  description = "The Generation of the Virtual Network gateway."
   type        = string
   default     = "Generation1"
+  nullable    = false
   validation {
     condition     = contains(["Generation1", "Generation2", "None"], coalesce(var.generation, "Generation1"))
     error_message = "Valid options are Generation1, Generation2 or None"
   }
 }
 
 variable "private_ip_address_enabled" {
-  description = "Should private IP be enabled on this gateway for connections?"
+  description = "Controls whether the private IP is enabled on the gateway."
+  default     = false
+  nullable    = false
   type        = bool
 }
 
+variable "zones" {
+  description = <<-EOF
+  After provider version 3.x you need to specify in which availability zone(s) you want to place IP.
+
+  For zone-redundant with 3 availability zones in current region value will be:
+  ```["1","2","3"]```
+  EOF
+  default     = null
+  type        = list(string)
+  validation {
+    condition = var.zones == null || (var.zones != null ? (
+      length(var.zones) == 3 && length(setsubtract(var.zones, ["1", "2", "3"])) == 0
+    ) : true)
+    error_message = "No zones or all 3 zones are expected"
+  }
+}
+
 variable "ip_configuration" {
   description = <<-EOF
-  List of IP configurations - every object in the list contains attributes:
+  IP configurations.
+
+  List of available attributes of each IP configuration.
 
-  - name - name of the IP configuration
-  - create_public_ip - boolean value, true if public IP needs to be created
-  - public_ip_name - name of the public IP resource used, when there is no need to create new one
-  - private_ip_address_allocation - defines how the private IP address of the gateways virtual interface is assigned. Valid options are Static or Dynamic. Defaults to Dynamic.
-  - public_ip_standard_sku - defaults to `false`, when set to `true` creates a Standard SKU, statically allocated public IP, otherwise it will be a Basic/Dynamic one.
-  - subnet_id - the ID of the gateway subnet of a virtual network in which the virtual network gateway will be created.
+  - `name`                          - (`string`, required) name of the IP configuration
+  - `create_public_ip`              - (`bool`, required) - true if public IP needs to be created
+  - `public_ip_name`                - (`string`, required) name of the public IP resource used, when there is no need to create new one
```

Comment:
you're not consistens when it comes to line wrapping :) , see lines: 141-143, 195, 197, 202, 259, 297, 320, 323 and 324

Reply:
fixed